src/utils/dns_utils.cpp

```diff
@@ -69,6 +70,28 @@ std::string UnescapeInstanceName(const std::string &aName)
     return newName;
 }
 
+void CheckHostnameSanity(const std::string &aHostName)
+{
+    OTBR_UNUSED_VARIABLE(aHostName);
+
+    assert(aHostName.length() > 0);
+    assert(aHostName[aHostName.length() - 1] == '.');
```

Comment:
nit: use `aHostName.back()`

Reply:
Just realized that these methods aren't introduced in this PR. Feel free to leave them unchanged.

Reply:
fixed to use .back()